diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -455,11 +455,14 @@ if(EXISTS "${CMAKE_SOURCE_DIR}/iconv/lib/iconv.lib")
 	config_write("#define ICONV_CONST \n\n")
 endif()
 
-if(WIN32)
+if(ANDROID)
+	# Android NDK does not provide gssapi_krb5.
+	set(lib_NETWORK)
+elseif(WIN32)
 	set(lib_NETWORK ws2_32 crypt32)
-else(WIN32)
+else()
 	# TODO check libraries
 	set(lib_NETWORK gssapi_krb5)
-endif(WIN32)
+endif()
 
 if(OPENSSL_FOUND)
 	set(lib_NETWORK ${lib_NETWORK} ${OPENSSL_LIBRARIES})
diff --git a/src/dblib/CMakeLists.txt b/src/dblib/CMakeLists.txt
--- a/src/dblib/CMakeLists.txt
+++ b/src/dblib/CMakeLists.txt
@@ -3,17 +3,23 @@ if(WIN32)
 	set(win_SRCS winmain.c dblib.def dbopen.c)
 endif()
 
+if(ANDROID)
+	set(android_SRCS dbopen.c)
+endif()
+
 # TODO add dbopen.c if necessary
 add_library(sybdb SHARED
 	dblib.c dbutil.c rpc.c bcp.c xact.c dbpivot.c buffering.h
 	${win_SRCS}
+	${android_SRCS}
 )
 target_compile_definitions(sybdb PUBLIC DLL_EXPORT=1)
 add_dependencies(sybdb encodings_h)
 target_link_libraries(sybdb tds replacements tdsutils ${lib_NETWORK} ${lib_BASE})
 
 add_library(db-lib STATIC
 	dblib.c dbutil.c rpc.c bcp.c xact.c dbpivot.c buffering.h
+	${android_SRCS}
 )
 add_dependencies(db-lib encodings_h)
 target_link_libraries(db-lib tds replacements tdsutils ${lib_NETWORK} ${lib_BASE})
